src/tests/unittests/asn1.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * asn1.c: Unit tests for ASN1 parsers
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Author: Jakub Jelen <jjelen@redhat.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "torture.h"
+#include "libopensc/asn1.h"
+
+/*
+ * Test undefined behavior of too large parts of OID encoding
+ *
+ * The specification does not place any limits to these values, but they
+ * are internally in opensc stored as ints so it makes sense to reject
+ * the too-large onese for now, rather than causing undefined overflow.
+ *
+ * https://oss-fuzz.com/testcase-detail/5673497895895040
+ */
+static void torture_large_oid(void **state)
+{
+	/* 2.5.4.18446744073709551619 (The last part is 64 bit overflow) */
+	u8 data1[] = {0x55, 0x04, 0x82, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x03};
+	/* 2.18446744073709551621.4.3 (The second part is 64 bit overflow) */
+	u8 data2[] = {0x82, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x55, 0x04, 0x03};
```

Comment:
those are two test scenarios, shouldn't they be separate test cases?
why 18446744073709551619 (0x10000000000000003) and not 18446744073709551616 (0x10000000000000000)? same for 18446744073709551621

shouldn't we have a test with 18446744073709551615 (0xffffffffffffffff) to verify that they _do_ work?

Reply:
yes, for signed int 32 bit type, 0x7fffffff and 0x80000000 are the values we want to test as the one working and the one outside supported range

Reply:
Already in as `data3` to `data6`.